flake.nix

```diff
@@ -14,35 +14,32 @@
         inherit system;
       };
       python_check_app = pkgs.writeShellApplication {
-        name = "python-check";
+        name = "lint-python";
         runtimeInputs = with pkgs; [
           ruff
           python311
           python311Packages.mypy
         ];
         text = ''
-          ruff format --check .
-          ruff check .
-          mypy .
-        '';
-      };
-      python_fix_app = pkgs.writeShellApplication {
-        name = "python-fix";
-        runtimeInputs = [pkgs.ruff];
-        text = ''
-          ruff format .
-          ruff check --fix .
+          set +u
+          if [ "$1" == "fix" ]
```

Comment:
```suggestion
          if [ "''${1:-}" == "fix" ]
```

Reply:
Oooo nice! Done.

 What is the `''` for?

Reply:
It's for escaping $ in nix